packages/http-server/src/__tests__/server.oas.spec.ts

```diff
@@ -11,19 +11,86 @@ function checkErrorPayloadShape(payload: string) {
   expect(parsedPayload).toHaveProperty('detail');
 }
 
-describe.each([['petstore.oas2.json'], ['petstore.oas3.json']])('server %s', file => {
+async function instantiatePrism(specPath: string) {
+  const server = createServer({}, { components: {}, config: { mock: { dynamic: false } } });
+  await server.prism.load({
+    path: relative(process.cwd(), specPath),
+  });
+  return server;
+}
+
+describe('GET /pet?__server', () => {
   let server: IPrismHttpServer<{}>;
 
   beforeAll(async () => {
-    server = createServer({}, { components: {}, config: { mock: { dynamic: false } } });
-    await server.prism.load({
-      path: relative(process.cwd(), resolve(__dirname, '..', '..', '..', '..', 'examples', file)),
+    server = await instantiatePrism(resolve(__dirname, 'fixtures', 'templated-server-example.oas3.json'));
+  });
+
+  afterAll(() => server.fastify.close());
+
+  describe.each([['http://stoplight.io/api'], ['https://stoplight.io/api']])('valid server %s', serverUrl => {
+    it('returns 200', () => {
+      return expect(requestPetGivenServer(serverUrl)).resolves.toMatchObject({
+        statusCode: 200,
+      });
+    });
+  });
+
+  describe.each([
+    ['ftp://stoplight.io/api'],
+    ['https://stoplight.com/api'],
+    ['https://google.com/api'],
+    ['https://stopligt.io/v1'],
+  ])('invalid server %s', serverUrl => {
+    it('returns 404 and problem json payload', () => {
+      return expect(requestPetGivenServer(serverUrl)).resolves.toMatchObject({
+        statusCode: 404,
+        payload: expectedPayload(serverUrl),
+      });
+    });
+  });
+
+  const expectedPayload = (serverUrl: string) =>
+    `{"type":"https://stoplight.io/prism/errors#NO_SERVER_MATCHED_ERROR","title":"Route not resolved, no server matched.","status":404,"detail":"The base url ${serverUrl} hasn\'t been matched with any of the provided servers"}`;
```

Comment:
I'm wondering if there's a way to check out this stuff without using this templated string.